extensions/amp-user-notification/amp-user-notification.md

```diff
@@ -0,0 +1,127 @@
+<!--
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+### <a name="amp-user-notification"></a> `amp-user-notification`
+
+Displays a dismissable notification to the user. By supplying two URLs that
+get called before the notification is shown and after it is dismissed,
+it is possible to control per user as to whether the notification should
+be shown (using the `ampUserId` value).
+E.g. it could only be shown to users in certain geo locations or
+it could not be shown against after the user has dismissed it once.
+
+---
+
+#### Usage
+
+`amp-user-notification` requires 2 URLs which can be provided by
+the `data-show-if-href` and `data-dismiss-href` attributes. An `id` is required
+as multiple `amp-user-notification` elements are allowed and the
+id is used to differentiate them.
+
+To close `amp-user-notification` add a `on` attribute to a button with the
+following value scheme `on="event:idOfUserNotificationElement.dismiss"`
+(see example below). This user action also triggers the `POST` to the
+`data-dismiss-href` URL.
+
+Example:
+
+```html
+<amp-user-notification
+    layout=container
+    id="amp-user-notification1"
+    data-show-if-href="http://localhost:8000/api/notification"
+    data-dismiss-href="http://localhost:8000/api/notification">
+    This site uses cookies to personalize content.
+    <a href="">Learn more.</a>
+   <button on="tap:amp-user-notification1.dismiss">I accept</button>
+</amp-user-notification>
+```
+
+---
+
+#### Attributes
+
+**data-show-if-href** (Required)
+
+AMP will make a request to this URL to determine whether the
```

Comment:
And link CORS to https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS